engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSource.scala

```diff
@@ -31,28 +37,42 @@ trait FlinkSource[T] extends Source[T] {
 
 }
 
+/**
+  * Support for Context initialization. By default initContext maps raw source event to "input" variable in Context.
+  * @tparam T - type of event that is generated by this source. This is needed to handle e.g. syntax suggestions in UI
+  */
+trait SourceContextTransformation[T] { self: Source[T] =>
```

Comment:
I like this concept of separation of context transformation from technical details like retrieving data etc. but:
1. IMO it would be better to have implementation of this transformation next to declaration (types transformation). Take a look at `ContextTransformation` as a reference.
2. If it is flink sepcific maybe we should suffix it by `Flink`?

Reply:
3. Why it has `T`? Maybe when we make it used by composition it wont be needed anymore?

Reply:
Fixed. Added comments, and moved both definition and implementation of this transformation to `FlinkContextInitializer`. `T` is required by initContext to provide proper MapFunction[T, Context].